src/Analysis/Abstract/Caching/FlowInsensitive.hs

```diff
@@ -1,4 +1,4 @@
-{-# LANGUAGE GADTs, GeneralizedNewtypeDeriving, TypeOperators #-}
```

Comment:
Integrating precise ASTs into `semantic`, which currently assumes à-la-carte ASTs everywhere, was challenging, and I went down a few blind alleys, which resulted in me making a bunch of unrelated simplifications which I’ve left in. Apologies for the chaff in the review!